recipes/cocoyaxi/all/conanfile.py

```diff
@@ -0,0 +1,107 @@
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_libcurl": [True, False],
+        "with_openssl": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_libcurl": False,
+        "with_openssl": False,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_libcurl:
+            self.requires("libcurl/7.79.1")
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1l")
+
+    def build_requirements(self):
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
+            #  The OSX_ARCHITECTURES target property is now respected for the ASM language
+            self.build_requires("cmake/3.20.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self._source_subfolder)
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        if not self.options.shared:
+            self._cmake.definitions["FPIC"] = self.options.get_safe("fPIC", False)
+        runtime = self.settings.get_safe("compiler.runtime")
+        if runtime:
+            self._cmake.definitions["STATIC_VS_CRT"] = "MT" in runtime
+        self._cmake.definitions["WITH_LIBCURL"] = self.options.with_libcurl
+        self._cmake.definitions["WITH_OPENSSL"] = self.options.with_openssl
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def package(self):
+        self.copy("LICENSE.md", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "cocoyaxi")
+        self.cpp_info.set_property("cmake_target_name", "cocoyaxi::co")
+        self.cpp_info.names["cmake_find_package"] = "cocoyaxi"
+        self.cpp_info.names["cmake_find_package_multi"] = "cocoyaxi"
+        self.cpp_info.components["co"].names["cmake_find_package"] = "co"
+        self.cpp_info.components["co"].names["cmake_find_package_multi"] = "co"
+        self.cpp_info.components["co"].set_property("cmake_target_name", "cocoyaxi::co")
```

Comment:
As I said in previous PR, cocoyaxi doesn't export its target in a config file, do you pan to implement it so that we can indeed set those values in cocyaxi recipe?

Reply:
> This is difficult for me. I don't know how to write such a config file with cmake..

Sorry I've changed the code scope of my review, so the best for now is to not set anything regarding CMake targets in this recipe for the moment, unless someone can help you to implement this in your library.

Reply:
https://cmake.org/cmake/help/git-stage/guide/importing-exporting/index.html

Reply:
Thanks. I'll take some time to learn it.

Reply:
When I have some time, I can help. So maybe keep these CMake names in cocoyaxi recipe.

Reply:
> When I have some time, I can help. So maybe keep these CMake names in cocoyaxi recipe.

@SpaceIm Great! Thank you very much!

Reply:
I've started to work on your upstream CMakeLists to create a CMake config file providing `cocoyaxi::co` imported target, I'll submit a PR soon.

Reply:
Thanks!

Reply:
here we go https://github.com/idealvin/cocoyaxi/pull/206

Reply:
> here we go [idealvin/cocoyaxi#206](https://github.com/idealvin/cocoyaxi/pull/206)

@SpaceIm  Thank you very much!